consent/api/v1/views.py

```diff
@@ -157,6 +159,11 @@ def get(self, request):
         exists = consent_exists(user_id, course_id, enterprise_customer_uuid)
         provided = consent_provided(user_id, course_id, enterprise_customer_uuid)
         required = consent_required(user_id, course_id, enterprise_customer_uuid)
+        enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+        if not enterprise_customer.catalog_contains_course_run(request.user, course_id):
+            # Only mark consent as required if it's in the Enterprise-linked
+            # course catalog; not otherwise.
+            required = False
```

Comment:
To avoid calling `consent_required` unnecessarily in some cases then, we can add an else clause here and call it there